Rubberduck.Inspections/Inspector.cs

```diff
@@ -103,18 +131,34 @@ private void UpdateInspectionSeverity(Configuration config)
                 return results;
             }
 
-            private void WalkTrees(CodeInspectionSettings settings, RubberduckParserState state, IEnumerable<IParseTreeInspection> inspections)
+            private void WalkTrees(CodeInspectionSettings settings, RubberduckParserState state, IEnumerable<IParseTreeInspection> inspections, ParsePass pass)
             {
-                var listeners =
-                    inspections.Where(i => i.Severity != CodeInspectionSeverity.DoNotShow && !IsDisabled(settings, i))
-                        .Select(inspection =>
-                        {
-                            inspection.Listener.ClearContexts();
-                            return inspection.Listener;
-                        })
-                        .ToList();
+                var listeners = inspections
+                    .Where(i => i.Severity != CodeInspectionSeverity.DoNotShow
+                        && i.Pass == pass
```

Comment:
It's not the inspection that needs to know; it's the inspector, needs a way to determine which parse tree to inspect with which inspections. The Pass is just a property of an inspection then; most work off the CodePane pass and aren't affected by this in any way.